presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -379,11 +406,29 @@ public boolean supportClientTransaction()
         return roles.build();
     }
 
+    /**
+     * Extracts a specific header PRESTO_EXTRA_CREDENTIAL
+     *
+     * @param headers a MultivaluedMap string
+     *
+     * @return a Map with the {@link PRESTO_EXTRA_CREDENTIAL} properties
+     */
     private static Map<String, String> parseExtraCredentials(MultivaluedMap<String, String> headers)
     {
         return parseProperty(headers, PRESTO_EXTRA_CREDENTIAL);
     }
 
+    /**
+     * Returns a map string with parsed properties.
```

Comment:
"Map string" seems strange to me. A better description would be:
"Returns a map containing all Presto properties." (Check if this is really the purpose)